### PR TITLE
♻️ Refactor study list search by adding empty state and make kfId searchable

### LINKED ISSUE #470: Study list view page completely disappears if search returns no results
I go to https://kf-ui-data-tracker.kidsfirstdrc.org
I click Screen Shot 2019-09-18 at 4 08 16 PM to switch to the list view.
I type "abc" into the search box.
The entire page disappears.
Screen Recording 2019-09-18 at 4 12 30 PM

### LINKED ISSUE #474: Study search box doesn't search all fields
I click in the search box. I'm going to search for `SD_DYPMEHHF` which I can clearly see.
Screen Shot 2019-09-18 at 5 00 40 PM

And then I start typing, but then everything goes away.
Screen Shot 2019-09-18 at 5 00 45 PM

### PR DESCRIPTION
**Pull request by XuTheBunny**
**Improvements:**
- Make KfID for study searchable from the search box
- Add empty search stage for study grid view and table view
- Add snapshot test on study grid view and table view with the empty search state

**UI changes:**
Empty search stage for grid view and table view:
Before:
image
After:
image
Searching by kfId:
image

**UI reference:**
Study list page when the user doesn't  have any studies: (already implemented)
For ADMIN user:
image
For not ADMIN user:
image

Closes #474
Closes #470

### CODE REVIEW

File: src/components/StudyList/StudyList.js [hunk 1]
```diff
@@ -53,6 +53,8 @@ const StudyList = ({studyList, loading, activeView = 'grid', user}) => {
       obj =>
         (obj.node.name &&
           obj.node.name.toLowerCase().includes(searchString.toLowerCase())) ||
+        (obj.node.kfId &&
+          obj.node.kfId.toLowerCase().includes(searchString.toLowerCase())) ||
```
Comment (dankolbman) [inline src/components/StudyList/StudyList.js]: I think you could accomplish all this by concatenating all the fields, then seeing if it includes the string:
```
(obj.node.name + obj.node.shortName + obj.node.kfId).toLowerCase().includes(searchString.toLowerCase())
```
